### ISSUE TITLE
Migration can occur without anyone voting

### ISSUE THREAD

**Issue by CaeraDenoir** [shieldify-security/Adapters-SPP#9]
## Severity
HIGH
## Description
Changes to migration logic makes possible to migrate without anyone voting. This is due to the new require checking if enough time has passed since the migration vote passed, but the variable itself is equal to zero before being set to timestamp + lock time.
## Impact
This has a huge impact since not only it can cause an unwanted migration, but also bypass the time granted to users to unstake if they are not comfortable with the migration.
## Location of Affected Code
In AdapterV1 contract, the logic in the migrateStake function.
## Recommendation
Add the previous check 

'''if (successMigrated == false) {
            revert ErrorsLib.migrationVotePending();
        }'''

Inside migrateStake,and modify successMigrated to true when the vote reaches the condition in acceptMigrationDestination

**Comment by CaeraDenoir** [comment]
I have some formatting errors since I was reviewing the code on the phone, but I think the general idea is there

**Comment by PossumLabsCrypto** [comment]
Great finding, thanks!

This again targets centralisation risk, which generally is out of scope but after expressing the intent that we want to give a higher assurance to users that we can´t just migrate their stake out of nowhere, this is certainly worth acknowledgement.

For context, the last time the related issue was reported, it was judged as "low" because of the reasoning above.
https://github.com/shieldify-security/SPP-Adapters/issues/7#issuecomment-2073009428

The likelihood of this being a problem is practically 0% because it requires a malicious multi-sig. However, given that I changed the code explicitely to eliminate this trust assumption, a medium severity seems appropriate.

Your suggested fix would unfortunately not solve the issue of minting the NFT to the destination, so I implemented a different fix that prevents `executeMigration()` before votes have reached >50%:
https://github.com/PossumLabsCrypto/Adapters/commit/04db5cca6d62116b1e20fdbe6d4b63e3ab5285e6

**Comment by aktech297** [comment]
Please clarify if my understanding is not correct.

I do see a check inside the **migrateStake** function.

https://github.com/shieldify-security/Adapters-SPP/blob/1a56f6f9de712e39d67eb7eba288696519eaa2bd/src/AdapterV1.sol#L158-L160

I think, its not an issue.

**Comment by CaeraDenoir** [comment]
The issue is that the function executeMigration, which sets the variable that migrateStake checks to true, can be called even if there is no votes

**Comment by aktech297** [comment]
oh, i was looking at the function **migrateStake**  from your submission.

even in this case, the below check will not allow to execute the function right?

[AdapterV1.sol#L123-L127](https://github.com/shieldify-security/Adapters-SPP/blob/1a56f6f9de712e39d67eb7eba288696519eaa2bd/src/AdapterV1.sol#L123-L127)

```solidity
    function executeMigration() external isMigrating {
        /// @dev Ensure that the timelock has passed
        if (block.timestamp < migrationTime) {
            revert ErrorsLib.isTimeLocked();
        }
```

because the **migrationTime** would be zero and it is `if (block.timestamp < migrationTime) `, so revert

**Comment by CaeraDenoir** [comment]
If migration time is 0, block.timestamp is higher => does not revert

**Comment by PossumLabsCrypto** [comment]
Degrading this to low because of lacking review of the fix which caused additional expenses in report #13

**Comment by CaeraDenoir** [comment]
You are right, my bad. Initially I thought a >50% votes requirement was added when reading your answer so I didn't think there would be any issue. (There would be an issue actually because the time lock would have been able to be bypassed if someone who did not vote unstaked, since it would have not fixed it entirely either)

**Comment by shealtielanz** [comment]
Heyy @PossumLabsCrypto 
The statement here in #13 is wrong
> The check only revert if migrationTime is set -> block.timestamp < migrationTime this evaluates to false, we have also &&, migrationTime > 0, since the first one is evaluated to false we don’t even go to the second check and also it will evaluate to true only when migrationTime is set

This only happens in the scenario of || `or` statement.
Also I think it is unfair for issue #9 because it found the root cause of this issue first. Shieldfy rules states that the first to submit take the rewards

**Comment by zanderbytexyz** [comment]
Hi, @shealtielanz!
1.The statement isn't incorrect. If `migrationTime == 0`, neither `block.timestamp < migrationTime` nor `migrationTime > 0` will evaluate to true. This statement is true only if both conditions are true. You can simply test it yourself.
2. I'm not entirely sure why this issue is being compared to mine, because in this particular case, the affected area of code points to the migrateStake() function (twice in the report) and `executeMigration` is not mentioned anywhere. I did see comments mentioning `executeMigration`, but they were published several hours after my submission. However, I don't understand the actual issue here based on the report above.
3. My report was made after the team implemented the fix, which is linked here in the comments.

**Comment by shealtielanz** [comment]
Heyy @zanderbytexyz 
I had an over sight but given your explanation here I have come to reconsider my inclination

**Comment by CaeraDenoir** [comment]
Hi @shealtielanz & @zanderbytexyz ! 
Thanks @shealtielanz  for reviewing the issues and giving your opinion on this. 

There are two things to differentiate here. 

First of all, it is true that the fix did not fix the issue, but it is also true that the fix is not the one I recommended (It is also true that the fix I recommended would have bring another issue, but that's beside the point).

Secondly, the reason as to why both mine and @zanderbytexyz issues are being compared is due to having the same root cause, `being able to migrate bypassing the need of votes due to migrationTime being 0` .

To be honest, I do not really know how this should be evaluated, and I did not know I had to check if the fix had the issue solved or not. I understand where the sponsor is coming from to lower my finding due to my lack of review causing him additional expenses, but now that I think about it, it is true that it's the same issue, it just wasn't fixed.

I am not sure if I should tag anyone from Shieldify to clarify everything

**Comment by PossumLabsCrypto** [comment]
If you think it´s worth escalating, please do contact Shieldify.

As an auditor it should go without saying that you:
1. Provide a fix that actually fixes the issue and not introduces new ones
2. If not done 1. at least check the posted fix by the sponsor / contractor

If you do neither 1 nor 2 it´s negligence on your job and frankly, still assigning a low severity to your report is being generous on my part. If @zanderbytexyz didn´t check the fix - which you should have done - this bug would have entered production.

**Comment by CaeraDenoir** [comment]
I do believe you are correct and it was totally my fault. I just gave my opinion regarding what is technically Shieldify terms. Regardless, I do not think it is worth escalating since even if it is technically the same issue, doing so would be even more unprofessional, and it wouldn't be fair for anyone. 

I am grateful at the Low severity, and would like to apologize for both my bad fix recommendation, and my comment above.
